gym_electric_motor/state_action_processors/state_action_processor.py

```diff
@@ -0,0 +1,127 @@
+import gym_electric_motor as gem
+
+
+class StateActionProcessor(gem.core.PhysicalSystem, gem.core.RandomComponent):
```

Comment:
Wouldn't be a better name "PhysicalSystemWrapper" ?
States and actions can also be processed when wrapping the environment, so the current name doesn't seem distinctive.

Reply:
Yes states and action could theoretically also be altered in a classical wrapper around the environment. The big advantage of a wrapper around the physical system is, that these derived quantities (e.g. the magnitude of the current vector in AC motor environments) can be directly referenced, rewarded and displayed.

PhysicalSystemWrapper could also be an adequate name. @max-schenke: what do you think about that suggestion?

Reply:
For me it seems to be a little bit of a choice between the RL- or control-based perspective, where the kind of plant system is not of primary interest, and between a hardware-based perspective that centers the drive.

Since GEM is not a toolbox for general controller or RL tuning, but specifically for the simulation of drive applications, I would go with your suggestion and support the "PhysicalSystemWrapper".